fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/RequestedOrderingConstraint.java

```diff
@@ -34,14 +35,40 @@ public class RequestedOrderingConstraint implements PlannerConstraint<Set<Reques
 
     @Nonnull
     @Override
-    public Optional<Set<RequestedOrdering>> combine(@Nonnull final Set<RequestedOrdering> currentConstraint, @Nonnull final Set<RequestedOrdering> newConstraint) {
-        if (currentConstraint.containsAll(newConstraint)) {
+    public Optional<Set<RequestedOrdering>> combine(@Nonnull final Set<RequestedOrdering> currentConstraint,
```

Comment:
This is now done in a better way as requested orders are hopefully combined in a better way leading to less re-exploration.